crates/c/src/lib.rs

```diff
@@ -1121,7 +1133,24 @@ impl InterfaceGenerator<'_> {
 
             match &self.resolve.types[ty].name {
                 Some(name) => self.define_type(name, ty),
-                None => self.define_anonymous_type(ty),
+                None => {
+                    // skip `typedef handle_x handle_y` where `handle_x` is the same as `handle_y`
+                    let name = &self.gen.type_names[&ty];
+                    match self.resolve.types[ty].kind {
+                        TypeDefKind::Handle(Handle::Borrow(id))
+                        | TypeDefKind::Handle(Handle::Own(id)) => {
+                            let info = &self.gen.resources[&dealias(self.resolve, id)];
+                            let target_borrow_name = &info.borrow;
+                            let target_own_name = &info.own;
+                            if !(name == target_borrow_name || name == target_own_name) {
```

Comment:
While I agree this works, if you're ok I'd prefer to avoid string comparisons and instead use a more declarative id-based comparison which is ideally a little less error-prone, something like:

```patch
diff --git a/crates/c/src/lib.rs b/crates/c/src/lib.rs
index b6145f2..c54687c 100644
--- a/crates/c/src/lib.rs
+++ b/crates/c/src/lib.rs
@@ -1129,8 +1128,19 @@ impl InterfaceGenerator<'_> {
     }

     fn define_anonymous_type(&mut self, ty: TypeId) {
-        self.src.h_defs("\ntypedef ");
         let kind = &self.resolve.types[ty].kind;
+
+        if let TypeDefKind::Handle(handle) = kind {
+            let resource = match handle {
+                Handle::Borrow(id) | Handle::Own(id) => id,
+            };
+            let origin = dealias(self.resolve, *resource);
+            if origin == *resource {
+                return;
+            }
+        }
+
+        self.src.h_defs("\ntypedef ");
         match kind {
             TypeDefKind::Type(_)
             | TypeDefKind::Flags(_)
```

where comparing the result of `dealias` and the original ID show indicate whether it's needed or not